homeassistant/components/update/__init__.py

```diff
@@ -0,0 +1,243 @@
+            return await registration.updates_callback(hass)
+    except asyncio.TimeoutError:
+        _LOGGER.warning("Timeout while getting updates from %s", registration.domain)
+    except Exception:  # pylint: disable=broad-except
+        _LOGGER.exception("Error fetching info")
+    return None
+
+
+def _get_update_details(
+    update_data: UpdateData,
+    domain: str,
+    identifier: str,
+) -> UpdateDescription | None:
+    """Get an update."""
+    return next(
+        (
+            update
+            for update in update_data.updates.get(domain, [])
+            if update.identifier == identifier
+        ),
+        None,
+    )
+
+
+def _filtered_updates(update_data: UpdateData) -> list[dict]:
+    """Return a list of updates that are not skipped."""
+    return [
+        {
+            "domain": domain,
+            "identifier": description.identifier,
+            "name": description.name,
+            "current_version": description.current_version,
+            "available_version": description.available_version,
+            "changelog_url": description.changelog_url,
+            "icon_url": description.icon_url,
+        }
+        for domain, domain_data in update_data.updates.items()
+        if domain_data is not None
+        for description in domain_data
+        if f"{domain}_{description.identifier}_{description.available_version}"
+        not in update_data.skip
+    ]
+
+
+@websocket_api.websocket_command({vol.Required("type"): "update/info"})
+@websocket_api.async_response
+async def handle_info(
+    hass: HomeAssistant,
+    connection: websocket_api.ActiveConnection,
+    msg: dict,
+):
+    """Get pending updates from all platforms."""
+    update_data: UpdateData = hass.data[DOMAIN]
+
+    for domain, domain_data in zip(
+        update_data.registrations,
+        await asyncio.gather(
+            *(
+                get_integration_info(hass, registration)
+                for registration in update_data.registrations.values()
+            )
+        ),
+    ):
+        update_data.updates[domain] = domain_data
+
+    connection.send_result(msg["id"], _filtered_updates(update_data))
+
+
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "update/skip",
+        vol.Required("domain"): str,
+        vol.Required("identifier"): str,
+    }
+)
+@websocket_api.async_response
+async def handle_skip(
+    hass: HomeAssistant,
+    connection: websocket_api.ActiveConnection,
+    msg: dict,
+):
+    """Skip an update."""
+    update_data: UpdateData = hass.data[DOMAIN]
+    update_details = _get_update_details(update_data, msg["domain"], msg["identifier"])
+    if update_details is not None:
+        update_data.skip.add(
+            f"{msg['domain']}_{update_details.identifier}_{update_details.available_version}"
+        )
+        update_data.updates[msg["domain"]].remove(update_details)
+        await update_data.store.async_save(list(update_data.skip))
```

Comment:
Can we move this logic into the `UpdateData` class? And then also do store initialization there. That way there is a single class responsible for the store.

Reply:
this should just move to the integration init and trigger on stop/final write event.

Reply:
I would just do a delay_save with 60 seconds. If someone is ignoring things, it's a few in a row. 

I still prefer if the code for loading/keeping it in memory/mutating/writing data is contained in a single class. Makes it easy to reason about and you don't need to look across different files how data is mutated.